server/src/main/java/io/druid/server/AsyncQueryForwardingServlet.java

```diff
@@ -138,12 +138,9 @@ public void init() throws ServletException
   {
     super.init();
 
-    // separate client with more aggressive connection timeouts
-    // to prevent cancellations requests from blocking queries
+    // Note that httpClinetProvider is setup to return same HttpClient instance on each get() so
+    // it is same http client as that is used by parent ProxyServlet.
     broadcastClient = httpClientProvider.get();
-    broadcastClient.setConnectTimeout(CANCELLATION_TIMEOUT_MILLIS);
```

Comment:
If setting the request timeout also has an effect of timing out the socket connection itself if that takes too long, then it sounds fine to me to just forget about `setConnectTimeout`, since it's being set at request time anyway.